packages/itwinui-css/src/file-upload/file-upload.scss

```diff
@@ -62,15 +64,150 @@
     }
   }
 
-  &.iui-drag > .iui-content {
-    background-color: var(--iui-color-background-accent-muted);
-    border-color: var(--iui-color-border-accent);
-    color: var(--iui-color-text-accent);
-    visibility: visible;
-
-    > svg,
-    .iui-template-icon {
+  &.iui-drag {
+    .iui-file-card-icon > svg,
+    .iui-file-card-empty-icon > svg {
       fill: var(--iui-color-icon-accent);
     }
+
+    > .iui-content {
+      visibility: visible;
+    }
+
+    > .iui-content,
+    .iui-file-card {
+      background-color: var(--iui-color-background-accent-muted);
+      border-color: var(--iui-color-border-accent);
+      color: var(--iui-color-text-accent);
+
+      > svg,
+      .iui-template-icon {
+        fill: var(--iui-color-icon-accent);
+      }
+    }
+  }
+}
+
+@mixin iui-file-card {
+  @include iui-reset;
+  @include iui-surface;
+  line-height: normal;
+  display: flex;
+  flex-direction: row;
+  align-items: center;
+  padding: calc(var(--iui-size-s) * 0.5) var(--iui-size-s);
+  border-radius: var(--iui-border-radius-1);
+  margin: calc(var(--iui-size-s) * 0.5) var(--iui-size-s);
```

Comment:
Continuing this conversation from [above](https://github.com/iTwin/iTwinUI/pull/984#discussion_r1125097228):

> > Have you tested whether it works without `FileUpload`? we need to make it work
> > 
> > also, the `files`/`onFilesChange` props should be added to `FileUpload`. usually in compound components, the component at the very top should control the state
> 
> I was able to get the sizing of the two cards to match by adding a `margin` to `iui-file-card` and removing it if it had a parent of `iui-content`. I have also fixed the double-outline issue, but since the outline goes around the `margin` now, the `focus-within` still just isn't quite right :/
> 
> ![image](https://user-images.githubusercontent.com/78564221/222846127-039b06a9-ae3d-46bb-b11d-a10576da203d.png)

Why are you setting both `margin` and `padding`? For the outer nodes, you should only set the `padding`. The icon also doesn't need extra margins, everything should be handled in the parent. Doing this will make the behavior more predictable and avoid such issues with focus outlines.

Reply:
Paired with Adam and removed all margins.